test/SourceKit/Diagnostics/cancel_diags.swift

```diff
@@ -0,0 +1,7 @@
+// RUN: not %sourcekitd-test -req=diags %s -print-raw-response -id=diag -async -- %s == -cancel=diag 2>&1 | %FileCheck %s
+
+func foo(x: Invalid1, y: Invalid2) {
+    x / y / x / y / x / y / x / y
```

Comment:
Not the biggest fan of these tests (same with the CursorInfoTest cases). It's okay for this PR, but IMO we should add a wait either somewhere in type checking or just in `ASTBuildOperation` that continues once it's set. That would allow something like:
```
// RUN: not %sourcekitd-test -req=diags %s -print-raw-response -id=diag -async -wait -- %s == -cancel=diag == -continue
```

Could even have different sections we wait at, eg. before type checking, during, after, etc.